crate_universe/tests/cargo_integration_test.rs

```diff
@@ -0,0 +1,148 @@
+    )
+    .join("cargo_home");
+    env::set_var("CARGO_HOME", cargo_home.as_os_str());
+    fs::create_dir_all(&cargo_home)?;
+
+    println!("$RUSTC={}", rustc.display());
+    println!("$CARGO={}", cargo.display());
+    println!("$CARGO_HOME={}", cargo_home.display());
+
+    Ok((cargo, rustc))
+}
+
+#[test]
+fn feature_generator() {
+    // This test case requires network access to build pull crate metadata
+    // so that we can actually run `cargo tree`. However, RBE (and perhaps
+    // other environments) disallow or don't support this. In those cases,
+    // we just skip this test case.
+    use std::net::ToSocketAddrs;
+    if "github.com:443".to_socket_addrs().is_err() {
+        eprintln!("This test case requires network access. Skipping!");
+        return;
+    }
+
+    let (cargo, rustc) = setup_cargo_env().unwrap();
+
+    let scratch = tempfile::tempdir().unwrap();
+    let manifest_path = scratch.path().join("Cargo.toml");
+    let runfiles = runfiles::Runfiles::create().unwrap();
+    fs::copy(
+        runfiles
+            .rlocation("rules_rust/crate_universe/test_data/metadata/target_features/Cargo.toml"),
+        manifest_path,
+    )
+    .unwrap();
+
+    let splicing_manifest = scratch.path().join("splicing_manifest.json");
+    fs::write(&splicing_manifest, serde_json::to_string(&json!({
+        "manifests": {
+            runfiles.rlocation("rules_rust/crate_universe/test_data/metadata/target_features/Cargo.toml").to_string_lossy(): "//:test_input"
+        },
+        "direct_packages": {},
+        "resolver_version": "2"
+    })).unwrap()).unwrap();
+
+    let config = scratch.path().join("config.json");
+    fs::write(
+        &config,
+        serde_json::to_string(&json!({
+            "generate_binaries": false,
+            "generate_build_scripts": false,
+            "rendering": {
+                "repository_name": "target_feature_test",
+                "regen_command": "//crate_universe:cargo_integration_test"
+            },
+            "supported_platform_triples": [
+                "x86_64-apple-darwin",
+                "x86_64-pc-windows-msvc",
+                "x86_64-unknown-linux-gnu",
+            ]
+        }))
+        .unwrap(),
+    )
+    .unwrap();
+
+    splice(SpliceOptions {
+        splicing_manifest,
+        cargo_lockfile: Some(
+            runfiles.rlocation(
+                "rules_rust/crate_universe/test_data/metadata/target_features/Cargo.lock",
+            ),
+        ),
+        repin: None,
+        workspace_dir: None,
+        output_dir: scratch.path().join("out"),
+        dry_run: false,
+        cargo_config: None,
+        config,
+        cargo,
+        rustc,
+    })
+    .unwrap();
+
+    let metadata = serde_json::from_str::<Value>(
+        &fs::read_to_string(scratch.path().join("out").join("metadata.json")).unwrap(),
+    )
+    .unwrap();
+
+    assert_eq!(
+        metadata["metadata"]["cargo-bazel"]["features"]["wgpu-hal 0.14.1"],
```

Comment:
Can the functionality here also account for platform specific configurations specified in the `Cargo.toml` file?

e.g.
```toml
[target.'cfg(unix)'.dependencies.tokio]
features = ["fs"]
```

Reply:
Yes!
Added this as another test case to the integration test.